packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,519 @@
+        tip_percentage: 10,
+        nonce: Math.floor(Math.random() * 1000000),
+        signer_public_keys: [
+          {
+            key_type: 'EddsaEd25519',
+            key_hex: this.getRadixPrivateKey().publicKey().hex(),
+          },
+        ],
+        flags: {
+          use_free_credit: true,
+          assume_all_signature_proofs: false,
+          skip_epoch_check: true,
+        },
+      }
+      const transactionPreviewResponse: TransactionPreviewResponse =
+        await this.gatewayApiClient.transaction.innerClient.transactionPreview({
+          transactionPreviewRequest: transactionPreviewRequest,
+        })
+      const receipt = transactionPreviewResponse.receipt as {
+        fee_summary: {
+          execution_cost_units_consumed: number
+          finalization_cost_units_consumed: number
+          xrd_total_execution_cost: string
+          xrd_total_finalization_cost: string
+          xrd_total_royalty_cost: string
+          xrd_total_storage_cost: string
+          xrd_total_tipping_cost: string
+        }
+      }
+      const totalFees =
+        parseFloat(receipt.fee_summary.xrd_total_execution_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_finalization_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_royalty_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_storage_cost) +
+        parseFloat(receipt.fee_summary.xrd_total_tipping_cost)
+      const estimatedFees: Fees = {
+        type: FeeType.FlatFee,
+        [FeeOption.Average]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fast]: assetToBase(assetAmount(totalFees)),
+        [FeeOption.Fastest]: assetToBase(assetAmount(totalFees)),
+      }
+      return estimatedFees
+    } catch (error) {
+      throw new Error('Failed to calculate the fees')
+    }
+  }
+
+  getRadixNetwork(): number {
+    const network = this.getNetwork()
+    let networkId: number
+    switch (network) {
+      case Network.Mainnet:
+        networkId = NetworkId.Mainnet
+        break
+      case Network.Testnet:
+        networkId = NetworkId.InternalTestNet
+        break
+      case Network.Stagenet:
+        networkId = NetworkId.Stokenet
+        break
+      default:
+        networkId = NetworkId.Stokenet
+        break
+    }
+    return networkId
+  }
+
+  getPrivateKey(): Buffer {
+    const seed = getSeed(this.phrase)
+    const bip32 = BIP32Factory(ecc)
+    const node: BIP32Interface = bip32.fromSeed(seed)
+    if (!this.rootDerivationPaths) throw new Error('no root derivation paths defined')
+    const child: BIP32Interface = node.derivePath(this.rootDerivationPaths[this.getNetwork()])
+    if (!child.privateKey) throw new Error('child does not have a privateKey')
+    return child.privateKey
+  }
+
+  getRadixPrivateKey(): PrivateKey {
+    const privateKey = this.getPrivateKey()
+    const privateKeyBytes = Uint8Array.from(privateKey)
+    return new PrivateKey.Ed25519(privateKeyBytes)
```

Comment:
This function is using BIP32 to derive an Ed25519 private key which is invalid since BIP32 is primarily intended for Secp256k1 doesn't work for Ed25519. If there is only tooling support for BIP32 and not SLIP10 then I recommend that we just support Secp256k1. Some example code on this: https://github.com/radixdlt/wallet-compatible-derivation/blob/e27de534a305561b1d7a9bbf8bc6c03c066d5425/crates/wallet_compatible_derivation/src/account_path.rs#L144-L151

Reply:
let me check if there's anything for SLIP10

Reply:
@0xOmarA I found a library that helps. so I added another param to the Client constructor where you can specify the curve that you want to use. both curves are now supported